src/producer/index.js

```diff
@@ -14,8 +14,26 @@ module.exports = ({
   const sendMessages = createSendMessages({ cluster, partitioner })
   const logger = rootLogger.namespace('Producer')
 
+  /**
+   * @typedef {Object} TopicMessages
+   * @property {string} topic
+   * @property {Array} messages An array of objects with "key" and "value", example:
+   *                         [{ key: 'my-key', value: 'my-value'}]
+   *
+   * @typedef {Object} SendBatchRequest
+   * @property {Array<TopicMessages>} topicMessages
+   * @property {number} [acks=-1] Control the number of required acks.
+   *                           -1 = all replicas must acknowledge
+   *                            0 = no acknowledgments
+   *                            1 = only waits for the leader to acknowledge
+   * @property {number} [timeout=30000] The time to await a response in ms
+   * @property {Compression.Types} [compression=Compression.Types.None] Compression codec
+   *
+   * @param {SendBatchRequest}
+   * @returns {Promise}
+   */
   const sendBatch = async ({ acks, timeout, compression, topicMessages }) => {
-    if (topicMessages.some(({ topic }) => !topic)) {
+    if (topicMessages && topicMessages.length > 0 && topicMessages.some(({ topic }) => !topic)) {
```

Comment:
What happens if the user sends an empty array of `topicMessages`? My point with the previous comment was to validade if the array is empty, if we do that we can keep this check simpler

Reply:
Clearly I should not write code before having had coffee. Not sure what I was thinking. 🙈 a5a8f12